File: server_java_sample/src/main/java/SendSimpleSample.java
import cn.jiguang.common.resp.APIConnectionException;
import cn.jiguang.common.resp.APIRequestException;
import cn.jpush.api.JPushClient;
import cn.jpush.api.push.PushResult;
import cn.jpush.api.push.model.Platform;
import cn.jpush.api.push.model.PushPayload;
import cn.jpush.api.push.model.audience.Audience;
import cn.jpush.api.push.model.audience.AudienceTarget;
import cn.jpush.api.push.model.notification.Notification;

import java.util.HashMap;
import java.util.Map;

public class SendSimpleSample {

    public static void main(String[] args) throws APIConnectionException, APIRequestException {

        String appkey = args[0];
        String secret = args[1];
        String registrationId = args[2];

        JPushClient jpushClient = new JPushClient(secret, appkey);

        AudienceTarget target = AudienceTarget.registrationId(registrationId);
        Audience audience = Audience.newBuilder()
                .addAudienceTarget(target)
                .build();

        Map<String, String> extras = new HashMap<>();
        extras.put("key1", "val1");
        extras.put("key2", "val2");

        PushPayload payload = PushPayload.newBuilder()
                .setPlatform(Platform.android())
                .setAudience(audience)
                .setNotification(Notification.android("＊＊alert:1＊＊", "＊Simple＊Title", extras))
                .build();

        PushResult result = jpushClient.sendPush(payload);
        System.out.println(result);
    }
}

File: server_kotlin_sample/src/main/kotlin/SendSimpleSample.kt
import cn.jpush.api.JPushClient
import cn.jpush.api.push.model.Platform
import cn.jpush.api.push.model.PushPayload
import cn.jpush.api.push.model.audience.Audience
import cn.jpush.api.push.model.audience.AudienceTarget
import cn.jpush.api.push.model.notification.Notification

import java.util.HashMap


fun main(args: Array<String>) {

    val appkey = args[0]
    val secret = args[1]
    val registrationId = args[2]

    val jpushClient = JPushClient(secret, appkey)

    val target = AudienceTarget.registrationId(registrationId)
    val audience = Audience.newBuilder()
            .addAudienceTarget(target)
            .build()

    val extras = mapOf("key1" to "val1", "key2" to "val2")

    val payload = PushPayload.newBuilder()
            .setPlatform(Platform.android())
            .setAudience(audience)
            .setNotification(Notification.android("＊＊alert:1＊＊", "＊Simple＊Title", extras))
            .build()

    val result = jpushClient.sendPush(payload)
    println(result)
}
